Apps/W1/HybridGP/app/src/Migration/Support/WizardIntegration.codeunit.al

```diff
@@ -149,49 +144,86 @@ codeunit 4035 "Wizard Integration"
+            exit(false);
+
+        if not TaskScheduler.CanCreateTask() then
+            exit(false);
+
+        if not JobQueueEntry.WritePermission then
+            exit(false);
+
+        exit(true);
+    end;
+
     procedure ScheduleGPHistoricalSnapshotMigration()
     var
         GPConfiguration: Record "GP Configuration";
-        JobQueueEntry: Record "Job Queue Entry";
+        GPUpgradeSettings: Record "GP Upgrade Settings";
         HybridCloudManagement: Codeunit "Hybrid Cloud Management";
-        UserPermissions: Codeunit "User Permissions";
+        HybridGPManagement: Codeunit "Hybrid GP Management";
         TimeoutDuration: Duration;
         MaxAttempts: Integer;
         QueueCategory: Code[10];
         IsHandled: Boolean;
         OverrideTimeoutDuration: Duration;
         OverrideMaxAttempts: Integer;
+        SessionId: Integer;
+        FailoverToSession: Boolean;
     begin
-        if not UserPermissions.IsSuper(UserSecurityId()) then
-            exit;
-
-        if not TaskScheduler.CanCreateTask() then
-            exit;
-
-        if not JobQueueEntry.WritePermission then
-            exit;
-
-        TimeoutDuration := GetDefaultGPHistoricalMigrationJobTimeoutDuration();
+        TimeoutDuration := HybridGPManagement.GetDefaultJobTimeout();
         MaxAttempts := GetDefaultGPHistoricalMigrationJobMaxAttempts();
+        QueueCategory := HybridCloudManagement.GetJobQueueCategory();
 
         OnBeforeCreateGPHistoricalMigrationJob(IsHandled, OverrideTimeoutDuration, OverrideMaxAttempts);
         if IsHandled then begin
             TimeoutDuration := OverrideTimeoutDuration;
             MaxAttempts := OverrideMaxAttempts;
         end;
 
-        QueueCategory := HybridCloudManagement.GetJobQueueCategory();
+        FailoverToSession := not CanStartBackgroundJob();
 
-        CreateAndScheduleBackgroundJob(Codeunit::"GP Populate Hist. Tables",
-                TimeoutDuration,
-                MaxAttempts,
-                QueueCategory,
-                GPSnapshotJobDescriptionTxt);
+        if not FailoverToSession then begin
+            CreateAndScheduleBackgroundJob(Codeunit::"GP Populate Hist. Tables",
+                    TimeoutDuration,
+                    MaxAttempts,
+                    QueueCategory,
+                    GPSnapshotJobDescriptionTxt);
 
-        if GPConfiguration.Get() then begin
-            GPConfiguration."Historical Job Ran" := true;
-            GPConfiguration.Modify();
+            SendStartSnapshotResultMessage('', StrSubstNo(TelemetrySnapshotScheduledMsg, 'Job Queue'), false);
+            if GPConfiguration.Get() then begin
+                GPConfiguration."Historical Job Ran" := true;
+                GPConfiguration.Modify();
+            end;
         end;
+
+        if FailoverToSession then
+            if Session.StartSession(SessionId, Codeunit::"GP Populate Hist. Tables", CompanyName(), GPUpgradeSettings, TimeoutDuration) then begin
```

Comment:
Can we emit telemetry here before Session.StartSession?

Reply:
Will do